### PR TITLE
Enable inductor perf test on GCP A100 

### PR DESCRIPTION
**Pull request by weiwangmeta**
This PR tries to enable inductor performance nightly testing on A100 runner provided by GCP. Currently these GCP runners were created and maintained using scripts in https://github.com/fairinternal/pytorch-gha-infra/pull/82. 
For some reason the artifacts cannot (and does not need to) be uploaded to S3, so adding use-gha parameter to _linux-test.yml to avoid creating a new but mostly identical _linux-test.yml. 

Workflow test results: https://github.com/pytorch/pytorch/actions/runs/3642340544/jobs/6149691109 

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: .github/workflows/inductor-perf-test-nightly.yml
Comment (desertfire) [inline .github/workflows/inductor-perf-test-nightly.yml:4]: How do we test this PR? I think adding a trigger with the `ciflow/nightly` label would be helpful here.

File: .jenkins/pytorch/test.sh
Comment (desertfire) [inline .jenkins/pytorch/test.sh:284]: We are not monitoring the inference perf on our dashboard. This is nice to have if it doesn't add too much test time.

File: .jenkins/pytorch/test.sh
Comment (desertfire) [inline .jenkins/pytorch/test.sh:289]: Please leave a comment to mark that we are not checking accuracy here yet.

File: .github/workflows/inductor-perf-test-nightly.yml
Comment (weiwangmeta) [reply .github/workflows/inductor-perf-test-nightly.yml:4]: Yes, I will add the ciflow/nightly trigger. Also I just realized the name conflict "inductor", I should rename it to inductor-perf-test.

File: .jenkins/pytorch/test.sh
Comment (weiwangmeta) [reply .jenkins/pytorch/test.sh:284]: Let me remove the inference perf part, as the total huggingface testing time seems long (1h30m).

Comment (weiwangmeta) [reply .jenkins/pytorch/test.sh:284]: And I will also remove the inference perf part for timm and torchbench

**Comment by weiwangmeta** [comment]
https://github.com/pytorch/pytorch/actions/runs/3642340544/jobs/6149691109#step:10:682 has BlenderbotForCausalLM OOM (**huggingface** model)
https://github.com/pytorch/pytorch/actions/runs/3642340544/jobs/6149691462#step:10:1337 hf_GPT2_large has OOM (**torchbench**)
https://github.com/pytorch/pytorch/actions/runs/3642340544/jobs/6149691462#step:10:1505 hf_T5 has OOM (**torchbench**)

**Comment by xuzhao9** [comment]
I suggest we tune the GPU frequency like in:
```
https://github.com/pytorch/benchmark/blob/main/.github/workflows/userbenchmark-a100.yml#L37
```

File: .jenkins/pytorch/test.sh
Comment (desertfire) [inline .jenkins/pytorch/test.sh:287]: Some refactoring has been done upstream, https://github.com/pytorch/pytorch/blob/17941b12e0f0aeb018f5bb68502eca78bdf6f60b/.jenkins/pytorch/test.sh#L259. Please update correspondingly.

File: .jenkins/pytorch/test.sh
Comment (desertfire) [inline .jenkins/pytorch/test.sh:791]: Can we merge this `elif` with the following on and use `if` only for the last step to select test_inductor_huggingface or test_inductor_huggingface_perf?

File: .jenkins/pytorch/test.sh
Comment (weiwangmeta) [reply .jenkins/pytorch/test.sh]: Done, thanks for the reminder.

File: .jenkins/pytorch/test.sh
Comment (weiwangmeta) [reply .jenkins/pytorch/test.sh]: Done, thanks for the suggestion.

**Review by anijain2305 — APPROVED** [review]
Looks good to me. But, let's wait for @desertfire for final approval.

**Review by seemethere — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**
